lib/ansible/modules/cloud/ovirt/ovirt_disk.py

```diff
@@ -758,9 +760,34 @@ def main():
                 ret = disks_module.action(
                     action='export',
                     action_condition=lambda d: module.params['image_provider'],
-                    wait_condition=lambda d: d.status == otypes.DiskStatus.OK,
+                    wait_condition=lambda d: d.status == otypes.DiskStatpus.OK,
                     storage_domain=otypes.StorageDomain(name=module.params['image_provider']),
                 )
+        elif state == 'imported':
+            glance_service = connection.system_service().openstack_image_providers_service()
+            image_provider = search_by_name(glance_service, module.params['image_provider'])
+            images_service = glance_service.service(image_provider.id).images_service()
+            entity = search_by_name(images_service, module.params['name'])
+            if entity is None:
+                raise Exception("Image '%s' was not found." % module.params['name'])
+            images_service.service(entity.id).import_(
+                storage_domain=otypes.StorageDomain(
+                    name=module.params['storage_domain']
+                ) if module.params['storage_domain'] else None,
+                disk=otypes.Disk(
+                    name=module.params['name']
+                ),
+                import_as_template=False,
+            )
+            # Wait for disk to appear in system:
+            disk = disks_module.wait_for_import(
```

Comment:
Can't we use here:
```python
ret = disks_module.action(
                     action='import'
...,
```
?

Reply:
I think it wont work, because it's working with openstack_image_providers_service and not disks_service